Zend/tests/gh10168_3.phpt

```diff
@@ -25,5 +25,4 @@ new Test();
 --EXPECTF--
 object(Test)#1 (0) {
 }
-object(Test)#2 (0) {
-}
+NULL
```

Comment:
how can `self::$a = $this` return null?

Reply:
> Also make expression result of assignments consistent, containing the value of the variable from after the destructor has been executed.

However, double checking, this is not how tests 1 and 2 behave (it would make sense actually adding a dump there, instead of doing a comparison). I'll have to take a closer look tomorrow.